iconservice/iconscore/icon_score_engine.py

```diff
@@ -126,15 +127,11 @@ def _call(self,
 
         icon_score = self._get_icon_score(context, icon_score_address)
 
-        is_func_readonly = getattr(icon_score, '_IconScoreBase__is_func_readonly')
-        if func_name is not None and is_func_readonly(func_name):
-            context.func_type = IconScoreFuncType.READONLY
-        else:
-            context.func_type = IconScoreFuncType.WRITABLE
-
         converted_params = self._convert_score_params_by_annotations(icon_score, func_name, kw_params)
-        external_func = getattr(icon_score, '_IconScoreBase__external_call')
-        return external_func(func_name=func_name, arg_params=[], kw_params=converted_params)
+        context.set_func_type_by_icon_score(icon_score, func_name)
```

Comment:
1. Although you set none func_name, It works as set default value(writable) and It has func_name surely, because that engine has validator about that It has method field.

2. @soobokjin is going to make issue about that and change IconServiceEngine_class to static class

Reply:
It has function to validate whether func_name is validate in _IconScoreBase__call